AN-589 CI no longer logs in to Docker

aednichols opened the pull request:
### Description

Iteratively sever links to Vault so we can one day remove it entirely. Docker Hub has raised pull limits since the original [BT-143](https://broadworkbench.atlassian.net/browse/WX-143) story. We also have better workarounds available such as GCR mirror.

This PR also removes Life Sciences tests that use the old Docker login mechanism that is being turned off. GCP Batch uses a new mechanism via ~KMS~ Secret Manager and is not affected.

### Release Notes Confirmation

#### `CHANGELOG.md`
 - [ ] I updated `CHANGELOG.md` in this PR
 - [x] I assert that this change shouldn't be included in `CHANGELOG.md` because it doesn't impact community users

#### Terra Release Notes
 - [ ] I added a suggested release notes entry in this Jira ticket
 - [x] I assert that this change doesn't need Jira release notes because it doesn't impact Terra users

jgainerdewar reviewed (approved):
Probably worth adding a note to the ticket or this PR about why PAPI tests were moved as part of this change.

aednichols commented:
I added links & explanations to both PRs to the story. I also updated this PR's description to explain why these particular tests are being disabled at this time.
